packages/mattermost-redux/src/reducers/entities/threads.ts

```diff
@@ -75,6 +87,29 @@ export const threadsReducer = (state: ThreadsState['threads'] = {}, action: Gene
 
 export const threadsInTeamReducer = (state: ThreadsState['threadsInTeam'] = {}, action: GenericAction) => {
     switch (action.type) {
+    case PostTypes.POST_REMOVED: {
+        const post = action.data;
+        if (post.root_id) {
+            return state;
+        }
+
+        const teamId = Object.keys(state).
+            find((id) => state[id].indexOf(post.id) !== -1);
+
+        if (!teamId) {
+            return state;
+        }
+
+        const index = state[teamId].indexOf(post.id);
```

Comment:
Would there be any benefit to checking if index is -1 here?

Reply:
It wouldn't hurt but since we check it in the find I thought it would be redundant.

```js
const teamId = Object.keys(state).find((id) => state[id].indexOf(post.id) !== -1);
```

Reply:
Oh, I missed that it was coming from the state already. I assumed it was coming from the action for some reason.